XamarinCommunityToolkit/Views/SegmentedView/SegmentedView.shared.cs

```diff
@@ -0,0 +1,199 @@
+
+		public static BindableProperty ColorProperty = BindableProperty.Create(nameof(Color), typeof(Color), typeof(SegmentedView));
+		public static BindableProperty SelectedIndexProperty = BindableProperty.Create(nameof(SelectedIndex), typeof(int), typeof(SegmentedView), 0, propertyChanged: OnSegmentSelected);
+		public static BindableProperty DisplayModeProperty = BindableProperty.Create(nameof(DisplayMode), typeof(SegmentMode), typeof(SegmentedView));
+
+		public IList<string> Items { get; } = new LockableObservableListWrapper();
+
+		public static readonly BindableProperty ItemsSourceProperty =
+			BindableProperty.Create(nameof(ItemsSource), typeof(IList), typeof(SegmentedView), default(IList),
+									propertyChanged: OnItemsSourceChanged);
+
+		public static readonly BindableProperty SelectedItemProperty =
+			BindableProperty.Create(nameof(SelectedItem), typeof(object), typeof(SegmentedView), null, BindingMode.TwoWay,
+									propertyChanged: OnSelectedItemChanged);
+
+		public IList ItemsSource
+		{
+			get { return (IList)GetValue(ItemsSourceProperty); }
+			set { SetValue(ItemsSourceProperty, value); }
+		}
+
+		public object SelectedItem
+		{
+			get { return GetValue(SelectedItemProperty); }
+			set { SetValue(SelectedItemProperty, value); }
+		}
+
+		public SegmentMode DisplayMode
+		{
+			get { return (SegmentMode)GetValue(DisplayModeProperty); }
+			set { SetValue(DisplayModeProperty, value); }
+		}
+
+		public static readonly BindableProperty CornerRadiusProperty =
+			BindableProperty.Create(nameof(CornerRadius), typeof(CornerRadius), typeof(SegmentedView), new CornerRadius(6.0));
+
+		public CornerRadius CornerRadius
+		{
+			get { return (CornerRadius)GetValue(CornerRadiusProperty); }
+			set { SetValue(CornerRadiusProperty, value); }
+		}
+
+		BindingBase itemDisplayBinding;
+
+		public BindingBase ItemDisplayBinding
+		{
+			get => itemDisplayBinding;
+
+			set
+			{
+				if (itemDisplayBinding == value)
+					return;
+
+				OnPropertyChanging();
+				itemDisplayBinding = value;
+				ResetItems();
+				OnPropertyChanged();
+			}
+		}
+
+		static readonly BindableProperty displayProperty =
+			BindableProperty.Create("Display", typeof(string), typeof(SegmentedView), default(string));
+
+		string GetDisplayMember(object item)
+		{
+			if (ItemDisplayBinding == null)
+				return item.ToString();
+
+			//TODO: Fix this
+			//ItemDisplayBinding.Apply(item, this, displayProperty);
+			//ItemDisplayBinding.Unapply();
+
+			return (string)GetValue(displayProperty);
+		}
+
+		static void OnItemsSourceChanged(BindableObject bindable, object oldValue, object newValue)
+		{
+			((SegmentedView)bindable)?.OnItemsSourceChanged((IList)oldValue, (IList)newValue);
+		}
+
+		void OnItemsSourceChanged(IList oldValue, IList newValue)
+		{
+			var oldObservable = oldValue as INotifyCollectionChanged;
+			if (oldObservable != null)
+				oldObservable.CollectionChanged -= CollectionChanged;
```

Comment:
Can you apply this change in others?

```suggestion
			if (oldObservable is INotifyCollectionChanged oldObservable)
				oldObservable.CollectionChanged -= CollectionChanged;
```

Reply:
Done.